lib/rubocop/cop/minitest/lifecycle_hooks_order.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Minitest
+      # Checks that lifecycle hooks are declared in the order in which they will be executed.
+      #
+      # @example
+      #   # bad
+      #   class FooTest < Minitest::Test
+      #     def teardown; end
+      #     def setup; end
+      #   end
+      #
+      #   # good
+      #   class FooTest < Minitest::Test
+      #     def setup; end
+      #     def teardown; end
+      #   end
+      #
+      #   # bad (after test cases)
+      #   class FooTest < Minitest::Test
+      #     def test_something
+      #       assert foo
+      #     end
+      #     def setup; end
+      #     def teardown; end
+      #   end
+      #
+      #   # good
+      #   class FooTest < Minitest::Test
+      #     def setup; end
+      #     def teardown; end
+      #     def test_something
+      #       assert foo
+      #     end
+      #   end
+      #
+      #   # good (after non test case methods)
+      #   class FooTest < Minitest::Test
+      #     def do_something; end
+      #     def setup; end
+      #     def teardown; end
+      #   end
+      #
+      class LifecycleHooksOrder < Base
+        include MinitestExplorationHelpers
+        include RangeHelp
+        extend AutoCorrector
+
+        MSG = '`%<current>s` is supposed to appear before `%<previous>s`.'
+
+        REGULAR_METHOD_POSITION = 100
```

Comment:
It would be better if source commented why the value of `100` was chosen and set :-)

Reply:
Updated.